frontend/client/components/Proposal/index.tsx

```diff
@@ -93,8 +95,12 @@ export class ProposalDetail extends React.Component<Props, State> {
       bodyId,
     } = this.state;
     const showExpand = !isBodyExpanded && isBodyOverflowing;
+    const wrongProposal = proposal && proposal.proposalId !== this.props.proposalId;
 
-    if (!proposal) {
+    if (detailError) {
+      return <ExceptionPage code="404" desc="Could not find that proposal" />;
+    }
+    if (!proposal || wrongProposal) {
       return <Loader size="large" />;
```

Comment:
Nice to have, by no means needs to change this PR: Now that we don't do optimistic rendering anymore from the search results, it'd be nice if we had a skeleton loading page instead of a plain spinner. It's a little more jarring now.